lib/rouge/lexers/idris.rb

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Idris < RegexLexer
+      title "Idris"
+      desc "The Idris programming language (idris-lang.org)"
+
+      tag 'idris'
+      aliases 'idr'
+      filenames '*.idr'
+      mimetypes 'text/x-idris'
+
+      reserved = %w(
+        _
+        data class instance namespace
+        infix[lr]? let where of with type 
+        do if then else case in
+      )
+
+      ascii = %w(
+        NUL SOH [SE]TX EOT ENQ ACK BEL BS HT LF VT FF CR S[OI] DLE
+        DC[1-4] NAK SYN ETB CAN EM SUB ESC [FGRU]S SP DEL [abfnrtv\\\"'\&]
+      )
```

Comment:
See earlier comment.

Reply:
Addressed in https://github.com/rouge-ruby/rouge/pull/1464/commits/d62d75e39180b4c87f5f64fa122918acfa6a6fe5